gibbon-compiler/src/Gibbon/Language/Syntax.hs

```diff
@@ -250,7 +312,46 @@ deriving instance (Show (TyOf ex), Show ex, Show (ArrowTy (TyOf ex))) => Show (P
 deriving instance (Eq (TyOf ex), Eq ex, Eq (ArrowTy (TyOf ex))) => Eq (Prog ex)
 deriving instance (Ord (TyOf ex), Ord ex, Ord (ArrowTy (TyOf ex))) => Ord (Prog ex)
 deriving instance Generic (Prog ex)
-deriving instance (NFData (TyOf ex), NFData (ArrowTy (TyOf ex)), NFData ex, Generic (ArrowTy (TyOf ex))) => NFData (Prog ex)
+
+deriving instance
+         (NFData (TyOf ex), NFData (ArrowTy (TyOf ex)), NFData ex,
+          Generic (ArrowTy (TyOf ex))) =>
+         NFData (Prog ex)
+
+-------------------------------------------------------------------------------
+-- Module Bundles
+-- Before modules get bundled into a single program, they're stored as
+-- a tuple of the discrte Prog and it's import declarations
+-------------------------------------------------------------------------------
+
+
+data ProgModule ex = ProgModule String (Prog ex) [ImportDecl SrcSpanInfo]
+data ProgBundle ex = ProgBundle [ProgModule ex] (ProgModule ex)
```

Comment:
Would be awesome to add comments to individual fields, see an example here: https://kowainik.github.io/posts/haddock-tips#what-can-be-documented

Reply:
This is great, I'll go through and clean up what comments I have